src/analyzer/protocol/quic/QUIC.spicy

```diff
@@ -160,308 +153,302 @@ const Version1: uint32 = 0x00000001;
-
-  switch ( self.version ) {
-    VersionDraft22,
-    VersionDraft23,
-    VersionDraft24,
-    VersionDraft25,
-    VersionDraft26,
-    VersionDraft27,
-    VersionDraft28,
-    VersionDraft29,
-    VersionDraft30,
-    VersionDraft31,
-    VersionDraft32,
-    VersionDraft33,
-    VersionDraft34,
-    VersionFace001,
-    VersionFace002,
-    VersionFace00e,
-    VersionFace011,
-    VersionFace012,
-    VersionFace013,
-    Version1 -> v1: LongHeaderPacketV1(self);
-    Version2 -> v2: LongHeaderPacketV2(self);
-    * -> unknown: UnhandledVersion(self) {
-      throw "unhandled QUIC version 0x%x" % self.version;
+    var encrypted_offset: uint64;
+    var payload_length: uint64;
+    var client_conn_id_length: uint8;
+    var server_conn_id_length: uint8;
+    var is_initial: bool;
+    var is_retry: bool;
+
+    first_byte: bitfield(8) {
+        header_form: 7 &convert=cast<HeaderForm>(cast<uint8>($$));
+        fixed_bit: 6;
+        packet_type: 4..5;
+        type_specific_bits: 0..3 &convert=cast<uint8>($$);
+    };
+
+    version: uint32;
+    dest_conn_id_len: uint8 {
+        self.server_conn_id_length = $$;
     }
-  };
+    dest_conn_id: bytes &size=self.server_conn_id_length;
+    src_conn_id_len: uint8 {
+        self.client_conn_id_length = $$;
+    }
+    src_conn_id: bytes &size=self.client_conn_id_length;
+
+    switch (self.version) {
+        VersionDraft22,
+        VersionDraft23,
+        VersionDraft24,
+        VersionDraft25,
+        VersionDraft26,
+        VersionDraft27,
+        VersionDraft28,
+        VersionDraft29,
+        VersionDraft30,
+        VersionDraft31,
+        VersionDraft32,
+        VersionDraft33,
+        VersionDraft34,
+        VersionFace001,
+        VersionFace002,
+        VersionFace00e,
+        VersionFace011,
+        VersionFace012,
+        VersionFace013,
+        Version1 -> v1: LongHeaderPacketV1(self);
+        Version2 -> v2: LongHeaderPacketV2(self);
+        * -> unknown: UnhandledVersion(self) {
+            throw "unhandled QUIC version 0x%x" % self.version;
+        }
+    };
 };
 
 # A QUIC Frame.
 public type Frame = unit(header: LongHeaderPacket, from_client: bool, crypto: CryptoSinkUnit, crypto_sink: sink&) {
-  frame_type : uint8 &convert=cast<FrameType>($$);
-
-  # TODO: add other FrameTypes as well
-  switch ( self.frame_type ) {
-    FrameType::ACK1 -> a: ACKPayload(FrameType::ACK1);
-    FrameType::ACK2 -> b: ACKPayload(FrameType::ACK2);
-    FrameType::CRYPTO -> c: CRYPTOPayload(from_client) {
-      # Have the sink re-assemble potentially out-of-order cryptodata
-      crypto_sink.write(self.c.cryptodata, self.c.offset.result_);
-
-      # If the crypto unit has determined a valid length, ensure we
-      # don't attempt to write more bytes into the sink. If it doesn't,
-      # use 2000 bytes as an arbitrary limit required to observe the
-      # length of the contained Client Hello or Server Hello.
-      if ( crypto.length > 0 ) {
```

Comment:
Also mostly fly-by, I've come to like the spaces after and before the parenthesis. That's used in Spicy's and Zeek's C++  code and also in Zeek script, so I wonder how you'd feel about changing spicy-format to include them, too?

Reply:
That's fair, but I am not going to add it.

Reply:
There's nothing to fix here, all this is about choice in style. I personally dislike the style with extra spaces inside conditionals, and we do not have an expectation to use the same style across different languages anyway 🤷.